scipy/stats/_qmc.py

```diff
@@ -1011,12 +1065,25 @@ def _random(
         # important to have ``type(bdim) == int`` for performance reason
         sample = [van_der_corput(n, int(bdim), start_index=self.num_generated,
                                  scramble=self.scramble,
-                                 seed=copy.deepcopy(self.seed),
+                                 permutations=self._permutations[i],
                                  workers=workers)
-                  for bdim in self.base]
+                  for i, bdim in enumerate(self.base)]
 
         return np.array(sample).T.reshape(n, self.d)
 
+    def reset(self) -> Halton:
+        """Reset the engine to base state.
+
+        Returns
+        -------
+        engine : Halton
+            Engine reset to its base state.
+
+        """
+        super().reset()
+        self._initialize_permutations()
```

Comment:
Is this necessary once `QMCEngine.reset()` is fixed? Once that happens, in all cases, the `.rng` will be the same as it was before, creating the same permutations, as desired. We don't have to redo that work. `Sobol` doesn't recompute its scrambling.